test/test_encryption.py

```diff
@@ -471,15 +482,25 @@ def parse_auto_encrypt_opts(self, opts):
         kms_providers = opts['kms_providers']
         if 'aws' in kms_providers:
             kms_providers['aws'] = AWS_CREDS
-            if not any(AWS_CREDS.values()):
+            if not all(AWS_CREDS.values()):
                 self.skipTest('AWS environment credentials are not set')
+        if 'awsTemporary' in kms_providers:
+            kms_providers['aws'] = AWS_TEMP_CREDS
+            del kms_providers['awsTemporary']
+            if not all(AWS_TEMP_CREDS.values()):
+                self.skipTest('AWS Temp environment credentials are not set')
+        if 'awsTemporaryNoSessionToken' in kms_providers:
+            kms_providers['aws'] = AWS_TEMP_NO_SESSION_CREDS
+            del kms_providers['awsTemporaryNoSessionToken']
+            if not all(AWS_TEMP_NO_SESSION_CREDS.values()):
+                self.skipTest('AWS Temp environment credentials are not set')
         if 'azure' in kms_providers:
             kms_providers['azure'] = AZURE_CREDS
-            if not any(AZURE_CREDS.values()):
+            if not all(AZURE_CREDS.values()):
```

Comment:
What's the rationale behind changing these from `any` to `all`?

Reply:
They should have always been `all`, right? All of the variables need to be set in order to test, otherwise we'll get an error when one of the variables is unset. Hmmm as I type this out I realize this is a good thing. Using `any` will help us catch mistakes where one of the vars is not set by accident. Using `all` we won't notice because the test will just be skipped. 

I'll change it back to `any`.

Reply:
I think this is what we realized last time as well.